### PR TITLE
Enable MS.CA.UnitTests on Mono

### PR DESCRIPTION
**Pull request by jaredpar**
This makes the changes such that MS.CA.UnitTests can execute on Mono.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-compiler, @kg for review

File: src/Compilers/Core/CodeAnalysisTest/AnalyzerFileReferenceTests.cs [hunk 1]
```diff
@@ -247,6 +247,7 @@ public class TestAnalyzer : DiagnosticAnalyzer
                 // Test analyzer load failure.
                 var remoteTest = (RemoteAnalyzerFileReferenceTest)loadDomain.CreateInstanceAndUnwrap(typeof(RemoteAnalyzerFileReferenceTest).Assembly.FullName, typeof(RemoteAnalyzerFileReferenceTest).FullName);
                 var exception = remoteTest.LoadAnalyzer(analyzerFile.Path);
+                Console.WriteLine(exception);
```
Comment (jcouv) [inline src/Compilers/Core/CodeAnalysisTest/AnalyzerFileReferenceTests.cs]: >WriteLine [](start = 24, length = 9)

Is this intentional?

**Comment by kg** [comment]
This all seems right to me. Is my assumption correct that both of these platform requirements aren't reasonable/possible to address in mono? I don't have a good sense of what the Fusion stuff actually does.

File: src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs [hunk 2]
```diff
@@ -739,9 +739,10 @@ public void TestMergeChanges_BeforeAdjacent()
             Assert.Equal("o World", changes[0].NewText);
         }
 
-        [Fact]
+        [ConditionalFact(typeof(ClrOnly), Reason = "https://github.com/mono/mono/issues/10961")]
         public void TestMergeChanges_NoMiddleMan()
         {
+            try {
```
Comment (jcouv) [inline src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs]: >try { [](start = 12, length = 5)

nit: format of the try/catch and catch block

File: src/Compilers/Core/CodeAnalysisTest/AnalyzerFileReferenceTests.cs (hunk 1 above)
Comment (jaredpar) [reply src/Compilers/Core/CodeAnalysisTest/AnalyzerFileReferenceTests.cs]: Ugg. I thought I deleted all of the diagnostic code. Let me fix.

**Comment by jaredpar** [comment]
@kg 

> Is my assumption correct that both of these platform requirements aren't reasonable/possible to address in mono? 

That's my understanding based on slack discussions. All of these tests are essentially bypassing normal loading and calling directly into CLR for assembly information. Seems unlikely Mono is interested in replicating this. 

I did decide to use a different reason flag though for Fusion in case the decision changes. Makes it easier to track these down.

**Comment by jaredpar** [comment]
@jcouv PR feedback addressed

File: src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs [hunk 3]
```diff
@@ -135,10 +136,12 @@ public abstract class ExecutionCondition
     public static class ExecutionConditionUtil
     {
         public static bool IsWindows => Path.DirectorySeparatorChar == '\\';
+        public static bool IsUnix => Path.DirectorySeparatorChar == '/';
```
Comment (jcouv) [inline src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs]: >IsUnix [](start = 27, length = 6)

Should this be `!IsWindows`?

Comment (jaredpar) [reply src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs]: Sure.

File: src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs (hunk 2 above)
Comment (jaredpar) [reply src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs]: Sorry, missed that. deleting.

**Pull request merged**